### PR TITLE
Issue #7593: Update doc for translation

### PR DESCRIPTION
**Pull request by ShivamPandey00**
Issue: #7593
Example added.
``` mvn clean verify ``` Passed. 

Website image:

Example 1:

CLI o/p:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="Translation">
        <property name="fileExtensions" value="properties, translations"/>
        <property name="requiredTranslations" value="ja, fr"/>
    </module>
</module>
```

```
$ cat messages.properties
hello=Hello
cancel=Cancel
```
```
$ cat messages.translations
hello=Hallo
ok=OK
```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml messages.properties messages.translations
Starting audit...
[ERROR] D:\checkstyle:1: Properties file 'messages_ja.translations' is missing. [Translation]
[ERROR] D:\checkstyle:1: Properties file 'messages_fr.translations' is missing. [Translation]
[ERROR] D:\checkstyle:1: Properties file 'messages_ja.properties' is missing. [Translation]
[ERROR] D:\checkstyle:1: Properties file 'messages_fr.properties' is missing. [Translation]
Audit done.
Checkstyle ends with 4 errors.

```

Example 2:

CLI o/p:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
      "-//Checkstyle//DTD Check Configuration 1.3//EN"
      "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="Translation">     
        <property name="fileExtensions" value="properties, translations"/>
        <property name="baseName" value="^ButtonLabels.*$"/>
        <property name="requiredTranslations" value="ja, fr"/>
  </module>
</module>
```
```
$ cat ButtonLabels.properties
save=Save
cancel=Cancel
```
```
$ cat ButtonLabels.translations
save=Speichern
ok=OK
```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml ButtonLabels.properties ButtonLabels.translations
Starting audit...
[ERROR] D:\checkstyle:1: Properties file 'ButtonLabels_ja.properties' is missing. [Translation]
[ERROR] D:\checkstyle:1: Properties file 'ButtonLabels_fr.properties' is missing. [Translation]
[ERROR] D:\checkstyle:1: Properties file 'ButtonLabels_ja.translations' is missing. [Translation]
[ERROR] D:\checkstyle:1: Properties file 'ButtonLabels_fr.translations' is missing. [Translation]
Audit done.
Checkstyle ends with 4 errors.
```

Example 3:

CLI o/p:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="Translation">
        <property name="fileExtensions" value="properties, translations"/>
        <property name="requiredTranslations" value="ja,  fr"/>
    </module>
</module>
```
```
$ cat messages_fr.properties
greeting=Bonjour
name=Nom
```
```
$ cat messages_ja.properties
greeting=こんにちは
age=年齢
```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml messages_ja.properties messages_fr.properties
Starting audit...
[ERROR] D:\checkstyle:1: Properties file 'messages.properties' is missing. [Translation]
[ERROR] D:\checkstyle\messages_fr.properties:1: Key 'age' missing. [Translation]
[ERROR] D:\checkstyle\messages_ja.properties:1: Key 'name' missing. [Translation]
Audit done.
Checkstyle ends with 3 errors.

```
Example 4:

CLI o/p:

```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="Translation">
        <property name="fileExtensions" value="properties, translations"/>
        <property name="requiredTranslations" value="es, fr, de"/>
    </module>
</module>
```
```
$ cat messages_de.properties
hello=Hallo
bye=Tschüss
ok=OK
```
```
$ cat messages_es.properties
hello=Hola
bye=Adiós
ok=Aceptar
```
```
$ cat messages_fr.properties
hello=Bonjour
cancel=Annuler
```
```
$  java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml messages_es.properties messages_fr.properties messages_de.properties
Starting audit...
[ERROR] D:\checkstyle:1: Properties file 'messages.properties' is missing. [Translation]
[ERROR] D:\checkstyle\messages_de.properties:1: Key 'cancel' missing. [Translation]
[ERROR] D:\checkstyle\messages_es.properties:1: Key 'cancel' missing. [Translation]
[ERROR] D:\checkstyle\messages_fr.properties:1: Key 'bye' missing. [Translation]
[ERROR] D:\checkstyle\messages_fr.properties:1: Key 'ok' missing. [Translation]
Audit done.
Checkstyle ends with 5 errors.

```

### CODE REVIEW

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Please review the changes.

**Comment by romani** [comment]
@coderdeadpool , please share CLI for all examples.

example of violations:
 `[ERROR] [checkstyle] [ERROR] ...checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/messages_de.properties:1: Key 'DefaultLogger.auditFinished' missing. [Translation]`
`[ERROR] [checkstyle] [ERROR] /home/rivanov/java/github/romani/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle:1: Properties file 'messages_de.properties' is missing. [Translation]`

you need to show files and content of files to let user understand that.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java [hunk 1]
```diff
@@ -137,6 +137,20 @@
...
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * # Ok
+ * messages.properties
+ * messages.translations
+ *
+ * # Violation: extension without translations or properties
+ * messages.txt
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:148]: we do not report violation on files that not a target for Check, please prove by CLI that it works like this. I  sure it is not working like you show.

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by coderdeadpool** [comment]
@romani do i have to show files and content of files within the example i added ?

**Comment by romani** [comment]
yes

**Comment by coderdeadpool** [comment]
> 
> you need to show files and content of files to let user understand that.
We need to add the content of the file within the example, like below:
```
Consider the following properties file in the same directory:

#messages.properties
hello=Hello
cancel=Cancel

#messages_de.properties
hell=Hallo
ok=OK
        
The Translation check will find the typo in the German hello key, the missing ok key in the default resource file and the missing cancel key in the German resource file:

messages_de.properties: Key 'hello' missing.
messages_de.properties: Key 'cancel' missing.
messages.properties: Key 'hell' missing.
messages.properties: Key 'ok' missing.
```

**Comment by coderdeadpool** [comment]
> yes.
> 
> play with CLI and you will see what users need to see how it works

The CLI o/p i got is:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="Translation">
        <property name="baseName" value="^ButtonLabels.*$"/>
    </module>
</module>
```
```
$ cat ButtonLabels.properties ButtonLabels.translations MessageLabels.properties MessageLabels.translations
button.ok=OK
button.cancel=Cancel
button.ok=OK in translation
button.cancel=Cancel in translation
message.error=An error has occurred
message.warning=A warning has occurred
message.error=An error has occurred in translation
message.warning=A warning has occurred in translation
```
```
$ java -jar target/checkstyle-10.9.2-SNAPSHOT-all.jar -c config.xml ButtonLabels.properties ButtonLabels.translations MessageLabels.properties MessageLabels.translations
Starting audit...
Audit done.
```

**Comment by romani** [comment]
Investigate Check, analyse what helped you to understand it, share such examples with all in world

**Comment by coderdeadpool** [comment]
@romani Added CLI o/p in the PR description.

**Comment by romani** [comment]
@coderdeadpool , please minimize amount of files in examples, there should only 2 files for each example - it is enough.

It is hard to follow example with 7 files.

Please add examples directly to xdoc(xml) files to let me see all in webpage.

**Comment by coderdeadpool** [comment]
> 
> Example 3: Screenshot (68)
> 
> Example 4: Screenshot (69)

@romani In example 3 we need to add messages.properties and messages_en.properties to show "violaiton". And same for example 4

**Comment by romani** [comment]
Just reduce required translations.
By all means try to minimize example, to let user focus on what matters easily.

**Comment by coderdeadpool** [comment]
@romani Added changes.Please review the changes

**Comment by coderdeadpool** [comment]
@romani "checkstyle.checkstyle (Job spelling)" check is failing because of the words that are added in the file messages_de.properties, messages_es.properties, messages_fr.properties. But it was necessary to add those words to show the content of these files.

**Comment by coderdeadpool** [comment]
@romani updated Webpage images in the PR description

**Comment by coderdeadpool** [comment]
@romani How can i resolve this ? the error message says "If you are ok with the output of this run, you will need to
patch config/jsoref-spellchecker/whitelist.words ".

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java [hunk 2]
```diff
@@ -168,19 +234,33 @@
...
+ * message_info_de=Information
+ * </pre>
+ * <pre>
+ * # Ok
+ * messages_es.properties
+ * messages_fr.properties
+ * messages_de.properties
+ *
+ * # Violation: missing translation for es, fr and de
+ * messages.properties
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: move this Violation situation in separate `pre` tag to show that it will separate execution on same config.
in java code you can mix ok and violation. On different files it is not possible, so it should be visually different codeblocks (`pre` tags).

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: I think we need to make big exception for this Check for general rules on examples and make examples in format of
image

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: Consider the following properties file in the same directory:
```
some text
```
Validation results:
```
some violation text in brief form
```

Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: > I think we need to make big exception for this Check for general rules on examples and make examples in format of image

@romani that would be great. but we also have to show the Violation for the files which does not have the extention ".properties, .translations" and the same for all other examples. Please correct me if am wrong.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: If CLI place violation on them - yes.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: This is not clear.
Please read example in description , see my screenshot, one more time

Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: As per the example you gave.
its looks like this 
```
#messages.properties
hello=Hello
cancel=Cancel

#messages.translations
hello=Hallo
ok=OK
```
```
messages.properties: Key 'ok' missing.
messages.translations: Key 'cancel' missing.
```
is this will be okay ?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:262]: Looks good, let see how it looks like in html

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Added changes. Please review the changes.
Added webpage images in the PR description

**Comment by romani** [comment]
As you know how to generate website by GitHub, no need to share images.

It is better to share CLI output to confirm that described in examples is actually true.
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/4375aca_2023180919/config_misc.html#Translation_Examples

Please share CLI on second example, I don't think it works as you described.

**Comment by coderdeadpool** [comment]
@romani I am getting errors
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="Translation">
        <property name="fileExtensions" value=".properties, .translations"/>
        <property name="baseName" value="^ButtonLabels.*$"/>
        <property name="requiredTranslations" value="ja, de, fr, es"/>
    </module>
</module>

```
```
$ cat messages.properties
hello=Hello
cancel=Cancel

```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml -d messages.properties
Mar 22, 2023 1:27:59 PM com.puppycrawl.tools.checkstyle.Main runCli
FINE: Checkstyle debug logging enabled
Mar 22, 2023 1:27:59 PM com.puppycrawl.tools.checkstyle.Main runCli
FINE: Running Checkstyle with version: 10.9.3-SNAPSHOT
com.puppycrawl.tools.checkstyle.api.CheckstyleException: unable to parse configuration stream
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:324)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:266)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:380)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:338)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:195)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.net.MalformedURLException: Cannot invoke "String.length()" because "spec" is null
        at java.base/java.net.URL.<init>(URL.java:709)
        at java.base/java.net.URL.<init>(URL.java:570)
        at java.base/java.net.URL.<init>(URL.java:517)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:650)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startEntity(XMLEntityManager.java:1397)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startDTDEntity(XMLEntityManager.java:1363)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDTDScannerImpl.setInputSource(XMLDTDScannerImpl.java:257)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.dispatch(XMLDocumentScannerImpl.java:1152)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.next(XMLDocumentScannerImpl.java:1040)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:943)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
        at com.puppycrawl.tools.checkstyle.XmlLoader.parseInputSource(XmlLoader.java:81)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.parseInputSource(ConfigurationLoader.java:196)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:315)
        ... 5 more
Caused by: java.lang.NullPointerException: Cannot invoke "String.length()" because "spec" is null
        at java.base/java.net.URL.<init>(URL.java:615)
        ... 24 more
Main.errorCounter

```

**Comment by coderdeadpool** [comment]
> 
> Please share CLI on second example, I don't think it works as you described.

@romani Here are the CLI o/p for the second example:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
      "-//Checkstyle//DTD Check Configuration 1.3//EN"
      "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="Translation">
        <property name="baseName" value="^ButtonLabels.*$"/>
  </module>
</module>
```
```
$ cat ButtonLabels.properties
save=Save
cancel=Cancel
```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml ButtonLabels.properties
DefaultLogger.auditStarted
DefaultLogger.auditFinished
```
```
$ cat ButtonLabels.translations
save=Speichern
ok=OK
```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml ButtonLabels.translations
DefaultLogger.auditStarted
DefaultLogger.auditFinished
```

**Comment by romani** [comment]
You should execute on two files , not a single.

Translation works across several files, it is the only multi file Check.

Your CLI output must be same as output examples. Users should be able to copy config and file content from examples and receive "same" results as in examples

**Comment by coderdeadpool** [comment]
> You should execute on two files , not a single.
> 
> Translation works across several files, it is the only multi file Check.
> 
> Your CLI output must be same as output examples. Users should be able to copy config and file content from examples and receive "same" results as in examples

@romani I did that too, see the PR description, and the result is same.

**Comment by romani** [comment]
There should be single cli command that validate both files in same execution

**Comment by coderdeadpool** [comment]
> There should be single cli command that validate both files in same execution .
> 
> Put two files in same command, separate by space
@romani You are refering like this :
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
      "-//Checkstyle//DTD Check Configuration 1.3//EN"
      "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="Translation">
        <property name="baseName" value="^ButtonLabels.*$"/>
  </module>
</module>
```
```
$ cat ButtonLabels.properties ButtonLabels.translations
save=Save
cancel=Cancel
save=Speichern
ok=OK
```
```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml ButtonLabels.properties ButtonLabels.translations
DefaultLogger.auditStarted
DefaultLogger.auditFinished
```

**Comment by romani** [comment]
`cat` should separate commands for each file.
`java ...` command should be single for both files.

I do not see update in code.

**Comment by coderdeadpool** [comment]
@romani  Please elaborate, i am little lost here.

As you said 

> There should be single cli command that validate both files in same execution .
> 
> Put two files in same command, separate by space

i did   ```
$ java -jar target/checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml ButtonLabels.properties ButtonLabels.translations
DefaultLogger.auditStarted
DefaultLogger.auditFinished
```

> 
> I do not see update in code.
I didnt understand what do we have to do here. Please help me out.

**Comment by romani** [comment]
Github, generate website

**Comment by romani** [comment]
all outputs in PR description have no validation errors, there should be validation errors as you show in html docuemtnation

**Comment by coderdeadpool** [comment]
@romani there should be validation errors. but the only o/p am getting is ``` starting audit audit done ```, dont know why.


> 
> you always have to use `requiredTranslations` property to define 2 languages, in other case it have nothing to validate.

yes, i tries including that property too. but the results are the same.

**Comment by romani** [comment]
use released version of checkstyle:
```
rivanov@p5510:/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="Translation">
        <property name="fileExtensions" value=".properties, .translations"/>
        <property name="requiredTranslations" value="es, fr, de"/>
    </module>
</module>
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ cat messages.properties 
hello=Hello
cancel=Cancel
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ cat messages.translations 
hello=Hallo
ok=OK
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ java -jar checkstyle-10.9.1-all.jar -c config.xml messages.properties messages.translations 
Starting audit...
[ERROR] /var/tmp:1: Properties file 'messages_de.properties' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_fr.properties' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_es.properties' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_de.translations' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_fr.translations' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_es.translations' is missing. [Translation]
Audit done.
Checkstyle ends with 6 errors.

```

**Comment by coderdeadpool** [comment]
@romani Updated CLI o/p with expected output. Please check and let me know it ok or not.

**Comment by romani** [comment]
Looks ok, try to keep required translations to be two languages.
let's start integration of them to xdoc to see final result.

**Comment by coderdeadpool** [comment]
> Looks ok, try to keep required translations to be two languages. let's start integration of them to xdoc to see final result.

Yes we can use required translation to be french and japanese languages only, however it can be done only for example 1 and example 2. We have to use the languages that are included in given config checks for example 3 and example 4. Please correct me if i am wrong.

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Updated the config code. Please review it.
web page: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/69725c0_2023095301/config_misc.html#Translation

**Comment by romani** [comment]
in first example, CLI have 4 violation, but you share in html only two and not a messages that CLI produce.
Should be:
```
Properties file 'messages_ja.translations' is missing.
Properties file 'messages_fr.translations' is missing.
Properties file 'messages_ja.properties' is missing.
Properties file 'messages_fr.properties' is missing.
```

from xdoc:
> Note, that files with the same path and base name but which have different extensions will be considered as files that belong to different resource bundles. 

please move it before config, to explain reason of config.

**Comment by coderdeadpool** [comment]
> in first example, CLI have 4 violation, but you share in html only two and not a messages that CLI produce. Should be:
> 
> ```
> Properties file 'messages_ja.translations' is missing.
> Properties file 'messages_fr.translations' is missing.
> Properties file 'messages_ja.properties' is missing.
> Properties file 'messages_fr.properties' is missing.
> ```

Yes,i thought about that too, and i think cli is producing wrong validation error.
Like in example one, the there should be only two violation that is :
```
[ERROR] D:\checkstyle\messages.properties:1: Key 'ok' missing. [Translation]
[ERROR] D:\checkstyle\messages.properties:1: Key 'cancel' missing. [Translation]
```
this violation message is correct and should be there but its not. 
I am trying to figure out that, help me here.

**Comment by romani** [comment]
Not ok to merge, fix documentation to show how it works now

**Review by romani — CHANGES_REQUESTED** [review]
Must be done https://github.com/checkstyle/checkstyle/pull/12870#issuecomment-1482730295

**Comment by coderdeadpool** [comment]
> in first example, CLI have 4 violation, but you share in html only two and not a messages that CLI produce. Should be:
> 
> ```
> Properties file 'messages_ja.translations' is missing.
> Properties file 'messages_fr.translations' is missing.
> Properties file 'messages_ja.properties' is missing.
> Properties file 'messages_fr.properties' is missing.
> ```

@romani I think you have misunderstood the example in html that was attached in PR description, i didnt change the screenshot of webpage after the first changes
here you can see the updated  changes : https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/69725c0_2023095301/config_misc.html#Translation

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
> from xdoc:
> 
> > Note, that files with the same path and base name but which have different extensions will be considered as files that belong to different resource bundles.
> 
> please move it before config, to explain reason of config.

@romani Updated this change. Please review.

**Comment by romani** [comment]
@coderdeadpool , dear maintainer , I looked at doumentation that you provided for Transaltion Check and was interested to play with it, but when I copied example that is shared in web page but CLI works differently:
```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="Translation">
    <property name="fileExtensions" value="properties, translations"/>
    <property name="requiredTranslations" value="ja, fr"/>
  </module>
</module>

$ cat messages.properties 
hello=Hello
cancel=Cancel

$ cat messages.translations 
hello=Hallo
ok=OK

$ java -jar checkstyle-10.9.2-all.jar -c config.xml messages.properties messages.translations 
Starting audit...
[ERROR] /var/tmp:1: Properties file 'messages_ja.properties' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_fr.properties' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_ja.translations' is missing. [Translation]
[ERROR] /var/tmp:1: Properties file 'messages_fr.translations' is missing. [Translation]
Audit done.
Checkstyle ends with 4 errors.
```

image


I am new to checkstyle please suggest what is wrong.

**Comment by coderdeadpool** [comment]
@romani I am a beginner.  and not able to identify what will be the expected cli output here and how can we get that.
I request you to please help me with that.

**Comment by romani** [comment]
please "copy" what CLI does now to documentation, so in first example ,as I showed above should be 4 violations.
as I pointed in https://github.com/checkstyle/checkstyle/pull/12870#issuecomment-1482730295

**Comment by coderdeadpool** [comment]
@romani Got it, do we have to do same for second example too ?

**Comment by romani** [comment]
Yes, we just need to show what CLI is showing

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani updated the CLI o/p in example 1 and example 2.
changes ``` value ="ja, fr" ``` to ```value="fr". to shorten the o/p message in the html.
Please review  it.
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/cdb7bf9_2023192936/config_misc.html#Translation

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java [hunk 3]
```diff
@@ -128,28 +128,62 @@
...
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * #ButtonLabels.properties
+ * save=Save
+ * cancel=Cancel
+ *
+ * #ButtonLabels.translations
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:178]: Make it ButtonLabels_fr.properties

To show ability to catch missed translation.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java [hunk 4]
```diff
@@ -158,6 +192,22 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * #messages_ja.properties
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:198]: Show here file messages.properties with English content

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java [hunk 5]
```diff
@@ -168,19 +218,29 @@
...
- * As we can see from the configuration, the TranslationCheck was configured
- * to check an existence of 'es', 'fr' and 'de' translations. Let's assume that
- * we have the resource bundle:
+ * Example:
  * </p>
  * <pre>
- * messages_home.properties
- * messages_home_es_US.properties
- * messages_home_fr_CA_UNIX.properties
+ * #messages_es.properties
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:224]: Remove this example completely, as it duplicates previous, it is just a bit more translations.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java (hunk 3 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:178]: After changing it to ```ButtonLabels_fr.properties``` and ```ButtonLabels_fr.translation```
I am getting
```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
      "-//Checkstyle//DTD Check Configuration 1.3//EN"
      "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="Translation">
        <property name="fileExtensions" value="properties, translations"/>
        <property name="baseName" value="^ButtonLabels.*$"/>
        <property name="requiredTranslations" value="fr"/>
  </module>
</module>
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat ButtonLabels_fr.properties
hello=Bonjour
name=Nom
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat ButtonLabels_fr.translations
hello=Bonjour
cancel=Annuler
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$  java -jar checkstyle-10.9.3-SNAPSHOT-all.jar -c config.xml ButtonLabels_fr.properties ButtonLabels_fr.translations
Starting audit...
[ERROR] C:\Users\hp\Desktop:1: Properties file 'ButtonLabels.translations' is missing. [Translation]
[ERROR] C:\Users\hp\Desktop:1: Properties file 'ButtonLabels.properties' is missing. [Translation]
Audit done.
Checkstyle ends with 2 errors.
```

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:178]: Rename file to ButtonLabels_fr.properties

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:178]: `.translation` is only for one example.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java (hunk 5 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java (hunk 4 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java:202]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java (hunk 3 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java]: Done

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Updated changes.Please review
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/fdb8f68_2023093514/config_misc.html#Translation

**Review by romani — APPROVED** [review]
awesome work !!
